### DISCUSSION TITLE
Basic MQTT question

### DISCUSSION THREAD

**Discussion by SteffMUC** [tbnobody/OpenDTU#542] [Q&A] [answered]
Hi all,
have the OpenDTU now running for a week, and yes thanks for the great project - it runs great, really love it.
Based on an article in the German ct - I have now started to try the first dashboard in node red. However, it seems that I have basic problem: Somehow I cannot see anything in mqtt, which I do not really understand.

Setup:

- OpenDTU successfully running
- MQTT - according to web interface successfully connected, interval set to 15sec

However I cannot see any mqtt messages from OpenDTU, neither in node red nor in dedicated mqtt client.
If I connect via debug client to my mosquito server, I can send test messages "hello world" for my topic "solar/" and can see it in node red without problems.

If I go to the console in OpenDTU I cannot see ANY mqtt log lines, only transfer messages to my Hoymiles. Should I be able to see any output of mqtt in the console ?

Any hint, what someone could do basically wrong here ? I am little bit clueless here, even after long search in forums.

Thanks
--Stefan

![opendtu_mqtt](https://user-images.githubusercontent.com/25563500/215321136-c56ca359-4ed0-45b7-8aab-d69fd1d4e167.jpg)

**Comment by tbnobody** [comment]
> If I go to the console in OpenDTU I cannot see ANY mqtt log lines, only transfer messages to my Hoymiles. Should I be able to see any output of mqtt in the console ?

Thats correct. There is no logging output. Are you using ACLs in your mqtt installation? If the user "mqttuser" does not have the correct permissions to write to a specific topic the message is just dropped. Thats protocol specific.

**Comment by SteffMUC** [comment, marked as answer]
Firstly, thanks for the fast reaction, really appreciate it.
No, I am not using ACLs in the mqtt installation.
Did start mqtt in verbose mode, at least now I see that openDTU is "delivering" content:
1675015661: Received PUBLISH from OpenDTU-953004 (d0, q0, r1, m0, 'solar/dtu/uptime', ... (5 bytes))
1675015661: Received PUBLISH from OpenDTU-953004 (d0, q0, r1, m0, 'solar/dtu/ip', ... (15 bytes))
1675015661: Received PUBLISH from OpenDTU-953004 (d0, q0, r1, m0, 'solar/dtu/hostname', ... (14 bytes))
1675015661: Received PUBLISH from OpenDTU-953004 (d0, q0, r1, m0, 'solar/dtu/rssi', ... (3 bytes))
Still investigating why I cannot "read" this in node red.
Started to test it with mosquitto_sub on the command line,
with "mosquitto_sub -t "#" -v -u "mqttuser" -P "****" I do get the messages:
solar/dtu/hostname OpenDTU-0E8AAC
solar/dtu/rssi -34
solar/114183335899/name TSOL M800
However, once I specify "solar", I do not getting anything:
mosquitto_sub -t "solar" -v -u "mqttuser" -P "****" 
Neither with "solar/" nor with "'solar"....
Any hints ?

**Reply by SteffMUC** [reply]
Finally solved it, stupid me, beginners mistake. 
You have to subribe to "solar/#", was not aware of it.
Apologies!
